pymodbus/datastore/store.py

```diff
@@ -174,13 +177,13 @@ def getValues(self, address, count=1):
         start = address - self.address
         return self.values[start : start + count]
 
-    def setValues(self, address, values):
+    def setValues(self, address: int, values: Collection):
         """Set the requested values of the datastore.
 
         :param address: The starting address
         :param values: The new values to be set
         """
-        if not isinstance(values, list):
+        if not isinstance(values, abc.Collection):
```

Comment:
I think collection is far too broad, you need to add test case to prove that it works.

Reply:
yes he did but not inside the datastore, he wanted it to define the datastore, which is quite different (and a very legal request).

Reply:
I re-read the original issue description twice.  It clearly discusses `setValues()`, and never mentions `__init__`.

Let's wait for the original submitter to clear this confusion.